core/src/main/java/org/infinispan/conflict/impl/DefaultConflictManager.java

```diff
@@ -0,0 +1,416 @@
+      final VersionRequest request;
+      synchronized (versionRequestMap) {
+         request = versionRequestMap.computeIfAbsent(key, k -> new VersionRequest(k, stateConsumer.isStateTransferInProgress()));
+      }
+
+      try {
+         return request.completableFuture.get();
+      } catch (InterruptedException e) {
+         Thread.currentThread().interrupt();
+         throw new CacheException(e);
+      } catch (ExecutionException e) {
+         if (e.getCause() instanceof CacheException)
+            throw (CacheException) e.getCause();
+
+         throw new CacheException(e.getCause());
+      } finally {
+         synchronized (versionRequestMap) {
+            versionRequestMap.remove(key);
+         }
+      }
+   }
+
+   @Override
+   public Stream<Map<Address, CacheEntry<K, V>>> getConflicts() {
+      return getConflicts(installedTopology);
+   }
+
+   private Stream<Map<Address, CacheEntry<K, V>>> getConflicts(LocalizedCacheTopology topology) {
+      if (stateConsumer.isStateTransferInProgress())
+         throw log.getConflictsStateTransferInProgress(cacheName);
+
+      if (!streamInProgress.compareAndSet(false, true))
+         throw log.getConflictsAlreadyInProgress();
+
+      try {
+         return StreamSupport
+               .stream(new ReplicaSpliterator(topology), false)
+               .filter(filterConsistentEntries());
+      } finally {
+         streamInProgress.compareAndSet(true, false);
+      }
+   }
+
+   @Override
+   public boolean isConflictResolutionInProgress() {
+      return streamInProgress.get();
+   }
+
+   @Override
+   public void resolveConflicts() {
+      resolveConflicts(installedTopology);
+   }
+
+   @Override
+   public void resolveConflicts(CacheTopology topology) {
+      LocalizedCacheTopology localizedTopology;
+      if (topology instanceof LocalizedCacheTopology) {
+         localizedTopology = (LocalizedCacheTopology) topology;
+      } else {
+         localizedTopology = distributionManager.createLocalizedCacheTopology(topology);
+      }
+      doResolveConflicts(localizedTopology);
+   }
+
+   private void doResolveConflicts(final LocalizedCacheTopology topology) {
+      final Set<Address> preferredPartition = new HashSet<>(topology.getCurrentCH().getMembers());
+
+      if (trace) log.tracef("Attempting to resolve conflicts.  All Members %s, Installed topology %s, Preferred Partition %s",
+            topology.getMembers(), topology, preferredPartition);
+
+      final Phaser phaser = new Phaser(1);
+      getConflicts(topology).forEach(conflictMap -> processConflict(phaser, conflictMap, topology, preferredPartition));
+      phaser.arriveAndAwaitAdvance();
+   }
+
+   private void processConflict(Phaser phaser, Map<Address, CacheEntry<K, V>> conflictMap,
+                                                LocalizedCacheTopology topology, Set<Address> preferredPartition) {
+      stateTransferExecutor.execute(() -> {
```

Comment:
nitpick: trivial statement + long lambda as a method doesn't look right; I would move the delegation to another executor to the forEach